### PR TITLE
Fix and reorganize overlay code.

### PR DESCRIPTION
**Pull request by brian-brazil**
Correctly handle skaters changing number, skaters
with no name, skaters with no number, ignore foexp (for now),
count penalties correctly when colouring, handle null logo.

Break down big functions into smaller more understandable parts,
and use the new WS functionality to simplify.

There's more to be done, but this is a big step in the right direction.

### CODE REVIEW

File: html/views/overlays/interactive/admin.js [hunk 1]
```diff
@@ -133,7 +132,7 @@ $('select#Skaters').change(function(e) {
 	team = $( 'option[value=' + v + ']', $t ).attr('data-team');
 	name = $( 'option[value=' + v + ']', $t ).attr('data-name');
 	tnam = WS.state['ScoreBoard.Team(' + team + ').AlternateName(overlay)'];
-	tnam = tnam ? tnam : WS.state['ScoreBoard.Team(' + team + ').Name'];
+	tnam = (tnam != "") ? tnam : WS.state['ScoreBoard.Team(' + team + ').Name'];
```
Comment (frank-weinberg) [inline html/views/overlays/interactive/admin.js]: Isn't this `null` if the AlternateName isn't set?

File: html/views/overlays/interactive/index.js [hunk 2]
```diff
@@ -1,38 +1,101 @@
...
+	WS.Register( ['ScoreBoard.Period(*).Jam(*).TeamJam(*).JamScore',
+			'ScoreBoard.Period(*).Jam(*).TeamJam(*).Lead',
+			'ScoreBoard.Period(*).Jam(*).TeamJam(*).Lost'], function(k,v) { jamData(k,v); } );
+
+	WS.Register( [
+			'ScoreBoard.Team(*).Skater(*).Name',
+			'ScoreBoard.Team(*).Skater(*).Number',
+			'ScoreBoard.Team(*).Skater(*).Flags' ], function(k,v) {
+				var me = '.RosterTeam' + k.Team + ' .Team' + k.Team + ' .Skater[data-skaterId=' + k.Skater + ']';
+				var mb = '.PenaltyTeam' + k.Team + ' .Team' + k.Team + ' .Skater[data-skaterId=' + k.Skater + ']';
```
Comment (frank-weinberg) [inline html/views/overlays/interactive/index.js:33]: What does `me`or `mb` mean? Can't we use more readable variable names? (This applies to much of the code here.)

File: html/views/overlays/interactive/index.js [hunk 3]
```diff
@@ -1,38 +1,101 @@
...
+		if (v == null) {
+			$('.Number-'+k.Penalty, sel).remove();
+			$(sel).attr('data-count', $('.Penalty', sel).length);
+			return;
+		}
+		ensureSkaterExists(k.Skater, k.Team);
+		createPenalty(sel, k.Penalty, v);
+	} );
+
+	WS.Register( 'ScoreBoard.Team(*).Color(*)', function(k,v) {
```
Comment (frank-weinberg) [inline html/views/overlays/interactive/index.js]: The function only acts on `overlay_bg` and `overlay_fg` - wouldn't it make more sense to only register those keys.

File: html/views/overlays/interactive/admin.js (hunk 1 above)
Comment (brian-brazil) [reply html/views/overlays/interactive/admin.js]: Hmm, thought I'd already undone that one.

File: html/views/overlays/interactive/index.js (hunk 2 above)
Comment (brian-brazil) [reply html/views/overlays/interactive/index.js:33]: This is all pre-existing, there's more changes to fix this up and I felt the PR was already big enough.

File: html/views/overlays/interactive/index.js (hunk 3 above)
Comment (brian-brazil) [reply html/views/overlays/interactive/index.js]: Done

File: html/views/overlays/interactive/index.js [hunk 4]
```diff
@@ -1,38 +1,102 @@
...
+			var ns,r;
+			var rule;
+			// chrome seems to like things in lowercase
+			var rd = '#sb .colourteam'+k.Team;
 
-	WS.Register( 'ScoreBoard.Team(1)', function(k,v) { teamData(1, k,v); } );
-	WS.Register( 'ScoreBoard.Team(2)', function(k,v) { teamData(2, k,v); } );
+			if(k.Color == 'overlay_bg') ns = 'background-color';
+			if(k.Color == 'overlay_fg') ns = 'color';
+			if(ns) {
```
Comment (frank-weinberg) [inline html/views/overlays/interactive/index.js]: This condition is now always true.

Comment (brian-brazil) [reply html/views/overlays/interactive/index.js]: Fixed

**Pull request merged**